src/driver_servo.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 <Admobilize>
+ * MATRIX Labs  [http://creator.matrix.one]
+ * This file is part of MATRIX Creator MALOS
+ *
+ * MATRIX Creator MALOS is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <iostream>
+
+#include "./driver_servo.h"
+#include "./src/driver.pb.h"
+#include "matrix_hal/everloop_image.h"
+#include "matrix_hal/gpio_control.h"
+
+#define INPUT 0
+#define OUTPUT 1
+#define PWM 1
+#define CLK_FRQ 200000000
+
+namespace matrix_malos {
+
+const bool kServoDriverDebugEnabled = false;
+
+bool ServoDriver::ProcessConfig(const DriverConfig& config) {
+
+  ServoParams servo(config.servo());
+
+  matrix_hal::WishboneBus bus;
```

Comment:
Do you need to do the wishbone initialization here?

See how the Everloop driver gets a reference to the SPI bus. You're doing it here for every ProcessConfig call and you could do it only once.

You can have "matrix_hal::GPIOControl *" in the .h as the everloop driver does.

You can also check the base class MalosWishboneBase in malos_wishbone_base.h. You need to define

    void SetupWishboneBus(matrix_hal::WishboneBus*) {}

as the everloop driver does.

You can follow the calls to SetupWishboneBus in malos.cpp.